src/ast_editor.ml

```diff
@@ -436,24 +437,34 @@ module Command = struct
   let _reveal_ast_node =
     let handler (instance : Extension_instance.t) ~textEditor ~edit:_ ~args:_ =
       let document = TextEditor.document textEditor in
-      let webview_opt =
-        let ast_editor_state = Extension_instance.ast_editor_state instance in
-        Ast_editor_state.find_webview_by_doc ast_editor_state
-          (TextDocument.uri document)
-      in
-      let offset =
-        let selection = Vscode.TextEditor.selection textEditor in
-        let position = Vscode.Selection.start selection in
-        TextDocument.offsetAt document ~position
+      let (_ : unit Promise.t) =
+        let open Promise.Syntax in
+        let+ webview =
+          let ast_editor_state = Extension_instance.ast_editor_state instance in
+          match
+            Ast_editor_state.find_webview_by_doc ast_editor_state
+              (TextDocument.uri document)
+          with
+          | Some _ as r -> Promise.return r
+          | None ->
+            let+ () = open_ast_explorer_handler textEditor in
+            Ast_editor_state.find_webview_by_doc ast_editor_state
+              (TextDocument.uri document)
+        in
+        let offset =
+          let selection = Vscode.TextEditor.selection textEditor in
+          let position = Vscode.Selection.start selection in
+          TextDocument.offsetAt document ~position
+        in
+        match webview with
+        | Some webview -> send_msg "focus" (Ojs.int_to_js offset) ~webview
```

Comment:
In a subsequent PR, consider wrapping all the `"focus"` calls into a function. It's best to wrap these junky untyped apis as early as possible and stick to our typed wrappers in the rest of the codebase.

Reply:
Yes, I have a branch actually with some ast explorer cleanups. Will send a PR at some point